thriftbp/server_middlewares_test.go

```diff
@@ -332,3 +333,36 @@ func TestPanicMiddleware(t *testing.T) {
 		}
 	})
 }
+
+func TestPrometheusMetricsMiddleware(t *testing.T) {
+	testCases := []struct {
+		name    string
+		wantErr thrift.TException
+		wantOK  bool
+	}{
+		{"bp error", thrift.NewTProtocolExceptionWithType(thrift.PROTOCOL_ERROR, thriftbp.WrapBaseplateError(errors.New("test"))), false},
+		{"application error", thrift.NewTApplicationException(thrift.UNKNOWN_METHOD, "unknown err mdsg"), false},
+		{"compile error", baseplate.NewError(), false},
+		{"success", nil, true},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
🔕 we are still not checking whether we emitted the correct prometheus metrics in the test, is that something doable in prometheus go library?

Reply:
You can start the test with `m.Reset()` and then `testutil.GatherAndCompare` or ToFloat64

https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/testutil#GatherAndCompare

I think we'll be wanting some of our own, better testing helpers over time... we had some nice ones before that could do things like measure deltas for a metric + labels across the test case, but it's probably overkill for this.

Reply:
@kylelemons I added some testutil checks, I used `collectAndCount` and `ToFloat64`, but I was having trouble `ToFloat64` for the histogram type. Do you know of a way to test the values for the historgram?